daemon/cmd/daemon_main.go

```diff
@@ -409,15 +409,27 @@ func InitGlobalFlags(logger *slog.Logger, cmd *cobra.Command, vp *viper.Viper) {
 	flags.String(option.NodeEncryptionOptOutLabels, defaults.NodeEncryptionOptOutLabels, "Label selector for nodes which will opt-out of node-to-node encryption")
 	option.BindEnv(vp, option.NodeEncryptionOptOutLabels)
 
-	flags.Bool(option.EnableEncryptionStrictEgress, false, "Enable encryption strict mode")
-	option.BindEnv(vp, option.EnableEncryptionStrictEgress)
+	flags.Bool(option.EnableEncryptionStrictMode, false, "Enable encryption strict mode")
+	flags.MarkDeprecated(option.EnableEncryptionStrictMode, "Please use --enable-encryption-strict-mode-egress instead. This option will be removed in v1.19")
+	option.BindEnv(vp, option.EnableEncryptionStrictMode)
 
-	flags.String(option.EncryptionStrictModeCIDR, "", "In strict-mode encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped")
+	flags.String(option.EncryptionStrictModeCIDR, "", "In strict-mode encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped.")
+	flags.MarkDeprecated(option.EncryptionStrictModeCIDR, "Please use --encryption-strict-egress-cidr instead. This option will be removed in v1.19")
 	option.BindEnv(vp, option.EncryptionStrictModeCIDR)
 
 	flags.Bool(option.EncryptionStrictModeAllowRemoteNodeIdentities, false, "Allows unencrypted traffic from pods to remote node identities within the strict mode CIDR. This is required when tunneling is used or direct routing is used and the node CIDR and pod CIDR overlap.")
+	flags.MarkDeprecated(option.EncryptionStrictModeAllowRemoteNodeIdentities, "Please use --encryption-strict-egress-allow-remote-node-identities instead. This option will be removed in v1.19")
 	option.BindEnv(vp, option.EncryptionStrictModeAllowRemoteNodeIdentities)
 
+	flags.Bool(option.EnableEncryptionStrictEgress, false, "Enable strict mode encryption enforcement for egress traffic")
+	option.BindEnv(vp, option.EnableEncryptionStrictEgress)
+
+	flags.String(option.EncryptionStrictEgressCIDR, "", "In strict-mode-egress encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped.")
+	option.BindEnv(vp, option.EncryptionStrictEgressCIDR)
+
+	flags.Bool(option.EncryptionStrictEgressAllowRemoteNodeIdentities, false, "Allows unencrypted traffic from pods to remote node identities within the strict mode CIDR. This is required when tunneling is used or direct routing is used and the node CIDR and pod CIDR overlap.")
```

Comment:
Just to check: how compatible is the new `EncryptionStrictIngress` with this?

Reply:
To expand a bit on this on why the two are unrelated: Egress strict mode requires the user to set a `EncryptionStrictEgressCIDR` - i.e. a CIDR which classified all traffic to be encrypted. Since node IPs might also be part of that CIDR (assuming _no_ node-to-node encryption), we want to allow traffic to node IPs to be unencrypted (even though they are part of of the strict mode CIDR).

For ingress strict mode, there is no such CIDR. Instead, the traffic is classified as "requiring encryption" if it comes out of the tunnel. Therefore the exception for remote nodes doesn't apply, since that traffic will never come out of the tunnel.